### PR TITLE
Explain why DataWrapper takes a separate shape

### PR DESCRIPTION
**Pull request by inducer**
cc @matthiasdiener

### CODE REVIEW

File: pytato/array.py [hunk 1]
```diff
@@ -1598,10 +1612,13 @@ class DataWrapper(InputArgumentBase):
...
             name: Optional[str],
             data: DataInterface,
-            shape: ShapeType,
+            shape: ShapeType = None,
             tags: TagsType = frozenset()):
         super().__init__(name, tags=tags)
 
+        if shape is None:
+            shape = data.shape
+
```
Comment (kaushikcfd) [inline pytato/array.py:1621]: Not sure about this. We have generally tried to avoid input validation in the type's constructor.

**Review by kaushikcfd — CHANGES_REQUESTED** [review]
Thanks! Looks good except one minor comment.

File: pytato/array.py (hunk 1 above)
Comment (inducer) [reply pytato/array.py:1621]: Fair point. I'll get rid of it.

Comment (inducer) [reply pytato/array.py]: 5795767

**Pull request merged**